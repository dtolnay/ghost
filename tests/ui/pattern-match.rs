#![feature(generic_const_items)]
#![allow(incomplete_features)]

use ghost::phantom;

#[phantom]
struct MyPhantom<T: ?Sized>;

fn main() {
    let phantom = MyPhantom::<u8>;

    match phantom {
        MyPhantom => {}
    }

    match phantom {
        MyPhantom::<u8> => {}
    }
}
