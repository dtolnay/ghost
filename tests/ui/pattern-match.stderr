error: constant of non-structural type `MyPhantom<u8>` in a pattern
  --> tests/ui/pattern-match.rs:13:9
   |
 6 | #[phantom]
   | ----------
   | |
   | `MyPhantom<u8>` must be annotated with `#[derive(PartialEq)]` to be usable in patterns
   | constant defined here
...
13 |         MyPhantom => {}
   |         ^^^^^^^^^ constant of non-structural type
   |
   = note: see https://doc.rust-lang.org/stable/std/marker/trait.StructuralPartialEq.html for details

error: constant of non-structural type `MyPhantom<u8>` in a pattern
  --> tests/ui/pattern-match.rs:17:9
   |
 6 | #[phantom]
   | ----------
   | |
   | `MyPhantom<u8>` must be annotated with `#[derive(PartialEq)]` to be usable in patterns
   | constant defined here
...
17 |         MyPhantom::<u8> => {}
   |         ^^^^^^^^^^^^^^^ constant of non-structural type
   |
   = note: see https://doc.rust-lang.org/stable/std/marker/trait.StructuralPartialEq.html for details
